packages/ERTP/src/paymentLedger.js

```diff
@@ -113,16 +115,17 @@ export const makePaymentLedger = (
       X`rights were not conserved: ${total} vs ${newTotal}`,
     );
 
-    // commit point
-    payments.forEach(payment => paymentLedger.delete(payment));
+    return fatalAssert.notThrows(() => {
```

Comment:
I think this woudl be more familiar to programmers as a try/catch pattern:

```
    try {
      // COMMIT POINT
      payments.forEach(payment => paymentLedger.delete(payment));

      const newPayments = newPaymentBalances.map(balance => {
        const newPayment = makePayment();
        paymentLedger.init(newPayment, balance);
        return newPayment;
      });
      return harden(newPayments);
    } catch (ex) {
      throw fatalAssertAfterCommit(ex);
    }
```

Reply:
In ZCF code we have other options that are still less syntactic overhead.

Reply:
> In ZCF code we have other options that are still less syntactic overhead.

What do you have in mind?

Reply:
> I think this woudl be more familiar to programmers as a try/catch pattern:

Agreed. Closed https://github.com/endojs/endo/pull/733

But still curious about 

> In ZCF code we have other options that are still less syntactic overhead.